### DISCUSSION TITLE
Trouble with Dynamic Route (v2)

### DISCUSSION THREAD

**Discussion by ivorpad** [remix-run/remix#6817] [General]
I'm currently working on a Remix application and I'm trying to create a dynamic route with the path `/post/b/:id/edit`. According to the Remix [documentation](https://remix.run/docs/en/1.18.1/file-conventions/route-files-v2), I've created a file named `post.b.$id.edit.tsx` in the `app/routes` directory. However, this doesn't seem to be working as expected.

When I try to navigate to a URL like `/post/b/123/edit`, the route doesn't seem to match.

**Comment by ivorpad** [comment]
I had two files named `post.b.$id.tsx` and `post.b.$id.edit.tsx`. It appears that `post.b.$id.tsx` was taking precedence over `post.b.$id.edit.tsx`, causing the routing issue.

I resolved this by renaming `post.b.$id.tsx` to `post.b.$id._index.tsx`. This change fixed the routing issue and now both routes are working as expected.

**Reply by kiliman** [reply]
By default, Remix will look for the longest common prefix as its "parent" layout since we don't have folder nesting to show the parent-child relationship.

The `_` suffix allows you to specify that this segment should not be treated as a parent layout.

```
post.b.$id.tsx
post.b.$id_.edit.tsx
```
